File: 01_basic/01_选择排序/Main.java
// 选择排序
public class Main {
    public static void main (String[] args) {
        int[] nums = new int[]{6, 4, 7, 2, 9, 3, 1, 5, 8};
        int len = nums.length;
        for (int i=0;i<len-1;i++){
            int minPos = i;

            for (int j=i+1;j<len ;j++ ) {
                if (nums[j]<nums[minPos]){
                    minPos = j;
                }

            }
            System.out.println("minPos:"+minPos);
            int tmp = nums[i];
            nums[i] =  nums[minPos];
            nums[minPos] = tmp;
        }

        for (int i=0; i<len; i++){
            System.out.print(nums[i]+",");
        }
    }
}

File: 01_basic/01_选择排序/main.py
def main():
    list1 = [6, 4, 7, 2, 9, 3, 1, 5, 8]
    print("list1:",list1)

    for i in range(len(list1)-1):
        # 找出当前最小值
        minPos = i
        for j in range(i+1,len(list1)):
            if list1[j] < list1[minPos]:
                minPos = j

        print(minPos)
        # 内容调换
        list1[i],list1[minPos] = list1[minPos],list1[i]

    print(list1)

if __name__ == "__main__":
    main()
